metadata_layer.py

```diff
@@ -1556,10 +1563,27 @@ def _key(classification):
 
         for link in self.links:
             if link.rel in Hyperlink.METADATA_ALLOWED:
+                original_resource = None
+                if link.original:
+                    rights_status = RightsStatus.lookup(_db, link.original.rights_uri)
+                    original_resource, ignore = get_one_or_create(
+                        _db, Resource, url=link.original.href,
+                    )
+                    original_resource.data_source = data_source
```

Comment:
Do we know that the original resource's data source is the same as the derivative?